pkg/virt-launcher/virtwrap/cmd-server/server.go

```diff
@@ -341,6 +341,60 @@ func (l *Launcher) GetGuestInfo(ctx context.Context, request *cmdv1.EmptyRequest
 	return response, nil
 }
 
+// GetUsers returns the list of active users on the guest machine
+func (l *Launcher) GetUsers(ctx context.Context, request *cmdv1.EmptyRequest) (*cmdv1.GuestUserListResponse, error) {
```

Comment:
Why is the error never returned here?

Reply:
It follows the same pattern as the rest of the file. The error is returned as part of the message.

Reply:
OK, but what I don't understand is then if the error is never returned, why is it declared as a return value which effectively is always `nil`? Can you point me towards why this has to be like this please?

Reply:
I do believe that the format of the command is dictated by the gRPC protocol used to communicate these functions to the virt-handler.

Therefore the code does not have another choice but to comply.

Reply:
I still don't understand why the code follows a pattern of returning an error embedded in a message even though there's a return value present that is much better suited to return it. Is the error message checked for the error somehow?

IMHO tthis should be cleaned up as this is a case of [stringly typed](https://wiki.c2.com/?StringlyTyped). But not in this PR then :-)

Reply:
Created #3127 for this.

Reply:
The grpc/protobuf implementation generates the func signature with err even though we don't declare it (see cmd.proto and cmd.pb.go). Also, it uses the error in case something goes wrong during the grpc call itself.
We, on the other hand, only use our own declared response type for reporting non-grpc related errors. It is stringly typed for sure, but IIRC it has always been this way, even in the pre-grpc implementation. If not, blame me, I introduced grpc a while ago ;) However, I don't say they is no room for improvement, just wanted give some more context :)